src/daemon/gnet/pool.go

```diff
@@ -220,16 +221,15 @@ func (pool *ConnectionPool) Run(q chan struct{}) {
 		conn, err := ln.Accept()
 		if err != nil {
 			// When Accept() returns with a non-nill error, we check the quit
-			// channel to see if we should continue or quit. If quit, then we quit.
+			// channel to see if we should continue or quit . If quit, then we quit.
 			// Otherwise we continue
 			select {
 			case <-pool.quit:
-				return
+				close(pool.ops)
+				return nil
 			default:
 				// without the default case the select will block.
-				logger.Error("%v", err)
-				close(q)
-				return
+				continue
```

Comment:
Before, upon error, it quit, now it tries again, is that ok? Also should log error?